### ISSUE TITLE
EmscriptenApplication keyboard handler doesn't send key event corresponding to text input events

### ISSUE THREAD

**Issue by CaseyCarlin** [mosra/magnum#637]
The[ keyboard handler in the emscripten backend](https://github.com/mosra/magnum/blob/75c25da6e5c065e785f411bd47feb007458ccae0/src/Magnum/Platform/EmscriptenApplication.cpp#L582) has this section:

```
/* If the key name is a single letter or a start of an UTF-8
   sequence, pass it to the text input event as well */
if(app.isTextInputActive() && key.size() == 1 || (key.size() >= 1 && UnsignedByte(key[0]) > 127)) {
    TextInputEvent e{*event, key};
    app.textInputEvent(e);
    return e.isAccepted();
}
```

The comment says "pass it to the text input event **as well**" but this conditional returns instead.

Because of this, when passing events to the Magnum ImGui integration, ImGui will only ever get a KeyDown event or a TextInput event; never both. Furthermore, if ImGui does not get a key event to correspond with a text event, keyboard shortcuts never fire inside Input widgets.

Based on the code comment above, it seems like `return e.isAccepted();` should be removed, and indeed in my local branch this does re-enable keyboard shortcuts in input widgets with no apparent ill effects.

**Comment by bz-next** [comment]
I had a similar problem at one point, and what worked for me was to make sure the code:

```
    if(ImGui::GetIO().WantTextInput && !isTextInputActive())
        startTextInput();
    else if(!ImGui::GetIO().WantTextInput && isTextInputActive())
        stopTextInput();
```

appeared in my draw function, after drawing the ImGUI widgets but before calling `_imgui.drawFrame();`

Don't know if that's helpful.

**Comment by mosra** [comment]
Oh, that's quite bad actually, I wonder how nobody came across that until now.

I cross-checked with the SDL2 and GLFW application implementations and they emit both events, so it makes sense to do that here as well. Did some more adjustments to make the accept status propagate from both the press and key events (because otherwise they'd get propagated further to the page / browser), and commited the fix as 536d4e75fe54b9f2b060b0e52f7af627b7a80d9b.

Thank you!

**Comment by CaseyCarlin** [comment]
Thank you as well!
